bob-common/src/mapper.rs

```diff
@@ -136,12 +136,21 @@ impl Virtual {
 
     pub fn vdisk_id_from_key(&self, key: BobKey) -> VDiskId {
         match self.distribution_func {
-            DistributionFunc::Mod => (key % self.vdisks.len() as u64)
+            DistributionFunc::Mod => (Self::get_vdisk_id_by_mod(key, self.vdisks.len()))
                 .try_into()
-                .expect("u64 to u32"),
+                .expect("usize to u32"),
         }
     }
 
+    fn get_vdisk_id_by_mod(key: BobKey, len: usize) -> usize {
+        let mut rem = 0;
+        for byte in key.iter() {
+            rem += *byte as usize % len;
+            rem %= len;
```

Comment:
Incorrect calculation of the remainder. Data written before the update is not readable because of this

Reply:
C# proper implementation as an example:
```
            long rem = 0;
            long byteMaxRem = 1;
            for (int i = 0; i < _keyBytes.Length; i++)
            {
                rem = (rem + _keyBytes[i] * byteMaxRem) % divisor;
                byteMaxRem = (byteMaxRem * 256) % divisor;
            }

            return (int)rem;
```

Reply:
This can be optimized by processing 4 bytes per iteration by converting to Int32